src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2307,7 +2307,14 @@ private BoundExpression BindOutDeclarationArgument(DeclarationExpressionSyntax d
                 Debug.Assert(localSymbol.DeclarationKind == LocalDeclarationKind.OutVariable);
                 if ((InConstructorInitializer || InFieldInitializer) && ContainingMemberOrLambda.ContainingSymbol.Kind == SymbolKind.NamedType)
                 {
-                    Error(diagnostics, ErrorCode.ERR_ExpressionVariableInConstructorOrFieldInitializer, declarationExpression);
+                    if (InConstructorInitializer)
```

Comment:
It looks like we're duplicating this check in the nested `if`. Is it worth instead saying something like:

```csharp
if (ContainingMemberOrLambda.ContainingSymbol.Kind == SymbolKind.NamedType)
{
    if (InFieldInitializer)
    {
          CheckFeatureAvailability(...)
    } else if (InConstructorInitializer)
    {
           Error(....)
    }
}
```

Reply:
> It looks like we're duplicating this check in the nested if. Is it worth instead saying something like:

This ```if``` is going away soon because the next step is to allow declarations in constructor initializers too.

---
In reply to: [162744123](https://github.com/dotnet/roslyn/pull/24292#discussion_r162744123) [](ancestors = 162744123)